application_sandbox/overlapping_frames/main.cpp

```diff
@@ -187,515 +219,460 @@ vulkan::VulkanGraphicsPipeline buildTrianglePipeline(
+            VK_IMAGE_ASPECT_COLOR_BIT,  // subresourceRange.aspectMask
+            0,                          // subresourceRange.baseMipLevel
+            1,                          // subresourceRange.levelCount
+            0,                          // subresourceRange.baseArrayLayer
+            1,                          // subresourceRange.layerCount
+        },
+    };
+
+    VkImageView raw_image_view;
+    LOG_ASSERT(
+        ==, data->logger(),
+        app->device()->vkCreateImageView(app->device(), &image_view_create_info,
+                                         nullptr, &raw_image_view),
+        VK_SUCCESS);
+    image_views.push_back(
+        vulkan::VkImageView(raw_image_view, nullptr, &app->device()));
+  }
+
+  return image_views;
 }
 
 containers::vector<vulkan::VkImageView> buildSamplerImageViews(
-    vulkan::VulkanApplication *app,
+    vulkan::VulkanApplication* app,
     const containers::vector<vulkan::ImagePointer>& images,
     const entry::EntryData* data) {
-    containers::vector<vulkan::VkImageView> image_views(data->allocator());
-    image_views.reserve(app->swapchain_images().size());
-
-    for(int index=0; index < images.size(); index++) {
-        auto sampler_image = images[index].get();
-
-        VkImageViewCreateInfo image_view_create_info {
-            VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO,  // sType
-            nullptr,                                   // pNext
-            0,                                         // flags
-            sampler_image->get_raw_image(),            // image
-            VK_IMAGE_VIEW_TYPE_2D,                     // viewType
-            app->swapchain().format(),                 // format
-            {
-                VK_COMPONENT_SWIZZLE_IDENTITY,  // components.r
-                VK_COMPONENT_SWIZZLE_IDENTITY,  // components.g
-                VK_COMPONENT_SWIZZLE_IDENTITY,  // components.b
-                VK_COMPONENT_SWIZZLE_IDENTITY,  // components.a
-            },
-            {
-                VK_IMAGE_ASPECT_COLOR_BIT,  // subresourceRange.aspectMask
-                0,                          // subresourceRange.baseMipLevel
-                1,                          // subresourceRange.levelCount
-                0,                          // subresourceRange.baseArrayLayer
-                1,                          // subresourceRange.layerCount
-            },
-        };
-
-        VkImageView raw_image_view;
-        LOG_ASSERT(==, data->logger(), app->device()->vkCreateImageView(
-            app->device(),
-            &image_view_create_info,
-            nullptr,
-            &raw_image_view
-        ), VK_SUCCESS);
-        image_views.push_back(vulkan::VkImageView(raw_image_view, nullptr, &app->device()));
-    }
-
-    return image_views;
+  containers::vector<vulkan::VkImageView> image_views(data->allocator());
+  image_views.reserve(app->swapchain_images().size());
+
+  for (int index = 0; index < images.size(); index++) {
```

Comment:
While we are making changes, can we switch this in 2 ways, 
1) size_t (since we are comparing with images.size(), which is a size_t). Prevents a possible(unlikely) infinite loop
2) `size_t i = 0` to be more idiomatic C++

Reply:
done